### PR TITLE
fix(SplitView): Clipping of right pane

### LINKED ISSUE #3174: Clipping is not applied to Right pane in SplitView
## Current behavior

Right pane in SplitView is not clipped in compact mode.

## Expected behavior

Right pane should be clipped in compact mode.

## How to reproduce it (as minimally and precisely as possible)

Try `SplitView` with Right pane placement and Compact display mode.

## Environment

Nuget Package: latest

Package Version(s): dev

Affected platform(s):

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): fixes #3174

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Right pane in `SplitView` was not properly clipped.

## What is the new behavior?

Right pane in `SplitView` is properly clipped.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
Thanks Martin! For the UI tests, you can create an automated test which validates that a plain color of the pane is what it's supposed to be (or not depending on the clip), using this:
https://github.com/unoplatform/uno/blob/d1574aa518292c4148018da9e166e14f2794f580/src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs#L235

**Comment by MartinZikmund** [comment]
@jeromelaban thanks, will do!

File: src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml [hunk 1]
```diff
@@ -2863,6 +2863,11 @@
 								</VisualState>
 								<VisualState x:Name="ClosedCompactRight">
 									<Storyboard>
+										<xamarin:ObjectAnimationUsingKeyFrames Storyboard.TargetProperty="Clip"
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml:2866]: Why is this necessary? I assume it's a workaround, but what for?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml:2866]: The same solution is for the left clip, but it was probably forgotten for the right clip. UWP animates `RectangleGeometry.Transform` instead, which is not yet supported

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml:2866]: Fair enough. Can you add a comment explaining the reason for it please - ideally linking to an issue for the `RectangleGeometry.Transform` limitation

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml:2866]: It is actually documented here - https://github.com/unoplatform/uno/blob/8dcc6f2d3aa5d57fd3ee96efe5fe73855a45a0da/src/Uno.UI/UI/Xaml/Controls/SplitView/SplitViewTemplateSettings.cs#L36

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml:2866]: Added #3238 to track this
